test/test_crud.py

```diff
@@ -42,43 +38,39 @@
 ]
 
 
-def run_playbook_vcr(module, extra_vars=None, extra_args=None, record=False):
-    # Assemble extra parameters for playbook call
-    if extra_args is None:
-        extra_args = []
+def run_playbook_vcr(module, extra_vars=None, record=False):
+    if extra_vars is None:
+        extra_vars = {}
+    limit = None
     if record:
         # Cassettes that are to be overwritten must be deleted first
         record_mode = 'once'
-        extra_args.extend(['--extra-vars', 'recording=true'])
+        extra_vars['recording'] = True
     else:
         # Never reach out to the internet
         record_mode = 'none'
         # Only run the tests (skip fixtures)
-        extra_args.extend(['--limit', '!fixtures'])
+        limit = '!fixtures'
 
     # Dump recording parameters to json-file and pass its name by environment
     test_params = {'test_name': module, 'serial': 0, 'record_mode': record_mode}
     with tempfile.NamedTemporaryFile('w', suffix='json', prefix='fam-vcr-') as params_file:
         json.dump(test_params, params_file.file)
         params_file.file.close()
         os.environ['FAM_TEST_VCR_PARAMS_FILE'] = params_file.name
-        return run_playbook(module, extra_vars=extra_vars, extra_args=extra_args)
+        return run_playbook(module, extra_vars=extra_vars, limit=limit)
 
 
-def run_playbook(module, extra_vars=None, extra_args=None):
+def run_playbook(module, extra_vars=None, limit=None):
     # Assemble parameters for playbook call
-    path = 'test/test_playbooks/{}.yml'.format(module)
-    playbook_opts = ['ansible-playbook', '-vvvv', path]
-    if extra_vars:
-        playbook_opts.extend(['--extra-vars', extra_vars])
-    if extra_args:
-        playbook_opts.extend(extra_args)
-
-    cli = PlaybookCLI(playbook_opts)
-    cli.parse()
-    return cli.run()
+    os.environ['ANSIBLE_CONFIG'] = os.path.join(os.getcwd(), 'ansible.cfg')
+    path = os.path.join(os.getcwd(), 'test/test_playbooks/{}.yml'.format(module))
+    run = ansible_runner.run(extravars=extra_vars, playbook=path, limit=limit, verbosity=4, inventory=os.path.join(os.getcwd(), 'test/inventory/hosts'))
+    return run
 
 
 @pytest.mark.parametrize('module', MODULES)
 def test_crud(module, record):
-    assert run_playbook_vcr(module, record=record) == 0
+    run = run_playbook_vcr(module, record=record)
+    print(run.stdout.read())
```

Comment:
Is this like file.read() ? Will it read until a buffer is filled?

Reply:
`run()` is blocking, `read()` is like a file, it should read everything.

Reply:
https://ansible-runner.readthedocs.io/en/latest/python_interface.html#runner-stdout